### PR TITLE
Improve DynamoDB Adapter

### PR DESCRIPTION
**Pull request by santitigaga**
Before submitting a pull request, please read
https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing

## Category
- [ ] Feature
- [x] Fix

## Checklist
- [x] The pull request is complete according to the [guide of contributing](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing)
- [x] Automated tests are written
- [x] The documentation is up-to-date
- [x] The pull request has a descriptive title that describes what has changed, and provides enough context for the changelog

### CODE REVIEW

File: src/main/resources/driven-adapter/dynamo-db/dynamo-db-template-adapter.java.mustache [hunk 1]
```diff
@@ -1,19 +1,39 @@
 package {{package}}.dynamodb;
 
-import {{package}}.dynamodb.helper.TemplateAdapterOperations;
+import co.com.bancolombia.dynamodb.helper.TemplateAdapterOperations;
```
Comment (juancgalvis) [inline src/main/resources/driven-adapter/dynamo-db/dynamo-db-template-adapter.java.mustache]: Should keep the dynamic root package [like](https://github.com/bancolombia/scaffold-clean-architecture/pull/218/files#:~:text=package%20%7B%7Bpackage-,%7D,-%7D.dynamodb.helper%3B)

**Review by juancgalvis — CHANGES_REQUESTED** [review]
Friend please fix the root package

File: src/main/resources/driven-adapter/dynamo-db/dynamo-db-template-adapter.java.mustache (hunk 1 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/dynamo-db/dynamo-db-template-adapter.java.mustache]: thanks

Comment (santitigaga) [reply src/main/resources/driven-adapter/dynamo-db/dynamo-db-template-adapter.java.mustache]: fixed

**Review by juancgalvis — APPROVED** [review]

File: src/main/resources/driven-adapter/dynamo-db-reactive/dynamo-db-template-adapter.java.mustache [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public DynamoDBTemplateAdapter(DynamoDbEnhancedAsyncClient connectionFactory, ObjectMapper mapper) {
+        /**
+         *  Could be use mapper.mapBuilder if your domain model implement builder pattern
+         *  super(repository, mapper, d -> mapper.mapBuilder(d,ObjectModel.ObjectModelBuilder.class).build());
+         *  Or using mapper.map with the class of the object model
+         */
+        super(connectionFactory, mapper, d -> mapper.map(d, Object.class /*domain model*/), "table_name");
+    }
+
+    public Mono<List<Object /*domain model*/>> getEntityBySomeKey(String key) {
```
Comment (juancgalvis) [inline src/main/resources/driven-adapter/dynamo-db-reactive/dynamo-db-template-adapter.java.mustache]: should return a single item, what do you think?

File: src/main/resources/driven-adapter/dynamo-db-reactive/modelEntity.java.mustache [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package {{package}}.dynamodb;
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.*;
+
+/* Enhanced DynamoDB annotations are incompatible with Lombok #1932
+         https://github.com/aws/aws-sdk-java-v2/issues/1932*/
+@DynamoDbBean
+public class ModelEntity{
```
Comment (juancgalvis) [inline src/main/resources/driven-adapter/dynamo-db-reactive/modelEntity.java.mustache]: It does not has the correct java format

File: src/main/resources/driven-adapter/dynamo-db-reactive/build.gradle.mustache [hunk 4]
```diff
@@ -0,0 +1,7 @@
+dependencies {
+    implementation project(':model')
+    implementation 'org.springframework:spring-context'
+    implementation 'software.amazon.awssdk:dynamodb-enhanced:2.17.134'
```
Comment (juancgalvis) [inline src/main/resources/driven-adapter/dynamo-db-reactive/build.gradle.mustache:4]: Should be added the sts dependency?

File: src/main/resources/driven-adapter/dynamo-db/config/dynamodb-config.java.mustache [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
+                .withCredentials(new ProfileCredentialsProvider("default"))
                 .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpoint, null))
                 .build();
     }
 
+    @Bean
+    @Profile({"dev", "cer", "pdn"})
+    public AmazonDynamoDB amazonDynamoDBAsync(@Value("${aws.dynamodb.threads}") String threads) {
+        return AmazonDynamoDBAsyncClientBuilder
+                .standard()
```
Comment (juancgalvis) [inline src/main/resources/driven-adapter/dynamo-db/config/dynamodb-config.java.mustache:29]: Should add web identity token provider or it is present by default?

File: src/main/resources/driven-adapter/dynamo-db-reactive/dynamo-db-template-adapter.java.mustache (hunk 2 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/dynamo-db-reactive/dynamo-db-template-adapter.java.mustache]: you are ok, I fix that.

File: src/main/resources/driven-adapter/dynamo-db-reactive/modelEntity.java.mustache (hunk 3 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/dynamo-db-reactive/modelEntity.java.mustache]: fixed

File: src/main/resources/driven-adapter/dynamo-db-reactive/build.gradle.mustache (hunk 4 above)
Comment (santitigaga) [reply src/main/resources/driven-adapter/dynamo-db-reactive/build.gradle.mustache:4]: added

**Pull request merged**